App doesn't start on emulator and close inmediately on an Elephone with Android 5.1

Hi everybody!

I have imported Etar-calendar in Android Studio 3.0.1 clonning it from the repository through import assistance tool. After that, in the console I've executed the following commands:

git submodules init
git submodule update

Then I've refreshed the gradle files but it's crashed because external projects detatimepicker, ex and timezonepicker are empty. So I have imported them clonning each one manually. Thanks to that, now the whole project builds succesfully, but when I try it on the emulator (Nexus 5 API 25), the app doesn't start and Android studio remains waiting for the emulator to come online. I've tried it several times, and once I have the message on the emulator that the app will close. Moreover, If I try to run the app in an Elephone with Android 5.1 API 22, I always have the message  that the application has stopped as soon as I click on its icon.

Has anyone bumped into this problem? Any idea?

Thanks!

While tha app works on the emulator, on my Elephone P6000 Pro appears this error trying to pase to int an empty string from the method getDaysPerWeek:

am start -n "inigo.murgui.com/com.android.calendar.AllInOneActivity" -a android.intent.action.MAIN -c android.intent.category.LAUNCHER
Connected to process 16523 on device elephone-p6000_pro-K01TJ5B0400027022
Capturing and displaying logcat messages from application. This behavior can be disabled in the "Logcat output" section of the "Debugger" settings page.
D/Proxy: setHttpRequestCheckHandler
D/PreferenceManager: inflateFromResource
D/PreferenceGroup: addPreference, preference.getKey() = null, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 0, this.getKey() = null
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = null
D/PreferenceGroup: addPreference, preference.getKey() = pref_theme, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 0, this.getKey() = pref_theme
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = pref_theme
W/Resources: Converting to string: TypedValue{t=0x10/d=0xfffffffe a=-1}
D/PreferenceGroup: addPreference, preference.getKey() = preferences_default_start, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 1, this.getKey() = preferences_default_start
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_default_start
D/PreferenceGroup: addPreference, preference.getKey() = preferences_hide_declined, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 2, this.getKey() = preferences_hide_declined
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_hide_declined
D/PreferenceGroup: addPreference, preference.getKey() = preferences_show_week_num, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 3, this.getKey() = preferences_show_week_num
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_show_week_num
D/PreferenceGroup: addPreference, preference.getKey() = preferences_week_start_day, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 4, this.getKey() = preferences_week_start_day
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_week_start_day
D/PreferenceGroup: addPreference, preference.getKey() = preferences_days_per_week, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 5, this.getKey() = preferences_days_per_week
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_days_per_week
D/PreferenceGroup: addPreference, preference.getKey() = preferences_home_tz_enabled, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 6, this.getKey() = preferences_home_tz_enabled
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_home_tz_enabled
D/PreferenceGroup: addPreference, preference.getKey() = preferences_home_tz, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 7, this.getKey() = preferences_home_tz
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_home_tz
D/PreferenceGroup: addPreference, preference.getKey() = preferences_clear_search_history, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 8, this.getKey() = preferences_clear_search_history
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_clear_search_history
D/PreferenceGroup: addPreference, preference.getKey() = preferences_alerts_category, this.getKey() = null
D/Preference: setOrder, old = 2147483647, new = 1, this.getKey() = preferences_alerts_category
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_alerts_category
D/PreferenceGroup: addPreference, preference.getKey() = preferences_alerts, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 0, this.getKey() = preferences_alerts
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_alerts
D/PreferenceGroup: addPreference, preference.getKey() = preferences_alerts_ringtone, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 1, this.getKey() = preferences_alerts_ringtone
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_alerts_ringtone
D/PreferenceGroup: addPreference, preference.getKey() = preferences_alerts_vibrate, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 2, this.getKey() = preferences_alerts_vibrate
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_alerts_vibrate
D/PreferenceGroup: addPreference, preference.getKey() = preferences_alerts_popup, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 3, this.getKey() = preferences_alerts_popup
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_alerts_popup
D/PreferenceGroup: addPreference, preference.getKey() = preferences_custom_snooze_delay, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 4, this.getKey() = preferences_custom_snooze_delay
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_custom_snooze_delay
D/PreferenceGroup: addPreference, preference.getKey() = preferences_default_snooze_delay, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 5, this.getKey() = preferences_default_snooze_delay
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_default_snooze_delay
D/PreferenceGroup: addPreference, preference.getKey() = preferences_default_reminder, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 6, this.getKey() = preferences_default_reminder
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_default_reminder
D/PreferenceGroup: addPreference, preference.getKey() = preferences_quick_responses, this.getKey() = preferences_alerts_category
D/Preference: setOrder, old = 2147483647, new = 7, this.getKey() = preferences_quick_responses
D/Preference: onParentChanged, mParentDependencyMet = true, disableChild = false, this.getKey() = preferences_quick_responses
D/ExtensionsFactory: No custom extensions.
D/ActivityThread: BIND_APPLICATION handled : 0 / AppBindData{appInfo=ApplicationInfo{13f71f7f inigo.murgui.com}}
V/ActivityThread: Handling launch of ActivityRecord{270b4b4c token=android.os.BinderProxy@19886795 {inigo.murgui.com/com.android.calendar.AllInOneActivity}}
D/ExtensionsFactory: AllInOneMenuExtensions not found in properties file.
V/ActivityThread: ActivityRecord{270b4b4c token=android.os.BinderProxy@19886795 {inigo.murgui.com/com.android.calendar.AllInOneActivity}}: app=com.android.calendar.CalendarApplication@1168349b, appName=inigo.murgui.com, pkg=inigo.murgui.com, comp={inigo.murgui.com/com.android.calendar.AllInOneActivity}, dir=/data/app/inigo.murgui.com-1/base.apk
D/FeatureProxyBase: FeatureProxyBase class constructor
D/MultiWindow: MultiWindowProxy constructor.
D/FeatureProxyBase: getService(), serviceName = multiwindow_service_v1
D/FeatureProxyBase: FeatureProxyBase class constructor
D/MultiWindow: MultiWindowProxy constructor.
D/FeatureProxyBase: getService(), serviceName = multiwindow_service_v1
D/FeatureProxyBase: FeatureProxyBase class constructor
D/MultiWindow: MultiWindowProxy constructor.
D/FeatureProxyBase: getService(), serviceName = multiwindow_service_v1
W/art: Before Android 4.1, method android.graphics.PorterDuffColorFilter android.support.graphics.drawable.VectorDrawableCompat.updateTintFilter(android.graphics.PorterDuffColorFilter, android.content.res.ColorStateList, android.graphics.PorterDuff$Mode) would have incorrectly overridden the package-private method in android.graphics.drawable.Drawable
D/AccessibilityManager: setStateLocked: wasEnabled = false, mIsEnabled = true, wasTouchExplorationEnabled = false, mIsTouchExplorationEnabled = false, wasHighTextContrastEnabled = false, mIsHighTextContrastEnabled = false
                        java.lang.Throwable: setStateLocked
                            at android.view.accessibility.AccessibilityManager.setStateLocked(AccessibilityManager.java:553)
                            at android.view.accessibility.AccessibilityManager.tryConnectToServiceLocked(AccessibilityManager.java:636)
                            at android.view.accessibility.AccessibilityManager.<init>(AccessibilityManager.java:226)
                            at android.view.accessibility.AccessibilityManager.getInstance(AccessibilityManager.java:206)
                            at android.view.View.setFlags(View.java:9843)
                            at android.view.ViewGroup.initViewGroup(ViewGroup.java:536)
                            at android.view.ViewGroup.<init>(ViewGroup.java:525)
                            at android.view.ViewGroup.<init>(ViewGroup.java:520)
                            at android.view.ViewGroup.<init>(ViewGroup.java:516)
                            at android.view.ViewGroup.<init>(ViewGroup.java:512)
                            at android.widget.FrameLayout.<init>(FrameLayout.java:119)
                            at com.android.internal.policy.impl.PhoneWindow$DecorView.<init>(PhoneWindow.java:2326)
                            at com.android.internal.policy.impl.PhoneWindow.generateDecor(PhoneWindow.java:3463)
                            at com.android.internal.policy.impl.PhoneWindow.installDecor(PhoneWindow.java:3849)
                            at com.android.internal.policy.impl.PhoneWindow.getDecorView(PhoneWindow.java:2042)
                            at android.support.v7.app.AppCompatDelegateImplV9.createSubDecor(AppCompatDelegateImplV9.java:379)
                            at android.support.v7.app.AppCompatDelegateImplV9.ensureSubDecor(AppCompatDelegateImplV9.java:328)
                            at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:289)
                            at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
                            at com.android.calendar.AllInOneActivity.onCreate(AllInOneActivity.java:354)
                            at android.app.Activity.performCreate(Activity.java:6092)
                            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1112)
                            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2481)
                            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2608)
                            at android.app.ActivityThread.access$800(ActivityThread.java:178)
                            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1470)
                            at android.os.Handler.dispatchMessage(Handler.java:111)
                            at android.os.Looper.loop(Looper.java:194)
                            at android.app.ActivityThread.main(ActivityThread.java:5637)
                            at java.lang.reflect.Method.invoke(Native Method)
                            at java.lang.reflect.Method.invoke(Method.java:372)
                            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959)
                            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754)
I/AppCompatViewInflater: app:theme is now deprecated. Please move to using android:theme instead.
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: inigo.murgui.com, PID: 16523
                  java.lang.RuntimeException: Unable to start activity ComponentInfo{inigo.murgui.com/com.android.calendar.AllInOneActivity}: java.lang.NumberFormatException: Invalid int: ""
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2534)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2608)
                      at android.app.ActivityThread.access$800(ActivityThread.java:178)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1470)
                      at android.os.Handler.dispatchMessage(Handler.java:111)
                      at android.os.Looper.loop(Looper.java:194)
                      at android.app.ActivityThread.main(ActivityThread.java:5637)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at java.lang.reflect.Method.invoke(Method.java:372)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754)
                   Caused by: java.lang.NumberFormatException: Invalid int: ""
                      at java.lang.Integer.invalidInt(Integer.java:138)
                      at java.lang.Integer.parseInt(Integer.java:358)
                      at java.lang.Integer.parseInt(Integer.java:334)
                      at java.lang.Integer.valueOf(Integer.java:525)
                      at com.android.calendar.Utils.getDaysPerWeek(Utils.java:693)
                      at com.android.calendar.AllInOneActivity.setMainPane(AllInOneActivity.java:1046)
                      at com.android.calendar.AllInOneActivity.initFragments(AllInOneActivity.java:773)
                      at com.android.calendar.AllInOneActivity.onCreate(AllInOneActivity.java:387)
                      at android.app.Activity.performCreate(Activity.java:6092)
                      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1112)
                      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2481)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2608) 
                      at android.app.ActivityThread.access$800(ActivityThread.java:178) 
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1470) 
                      at android.os.Handler.dispatchMessage(Handler.java:111) 
                      at android.os.Looper.loop(Looper.java:194) 
                      at android.app.ActivityThread.main(ActivityThread.java:5637) 
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at java.lang.reflect.Method.invoke(Method.java:372) 
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:959) 
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:754) 
Application terminated.

--- a/res/values-es/strings.xml
+++ b/res/values-es/strings.xml
@@ -302,7 +302,6 @@
   <string name="no_events">Ningún evento disponible</string>
   <string name="preferences_theme">Tema</string>
   <string name="preferences_light_theme">Claro</string>
-  <string name="preferences_days_per_week_default"></string>
   <string name="preferences_dark_theme">Oscuro</string>
   <string name="preferences_days_per_week_dialog">Días por semana</string>
   <string name="preferences_days_per_week_title">Días por semana</string>

